pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -113,7 +115,14 @@ func (c *TargetConfigController) sync(ctx context.Context, syncCtx factory.SyncC
 		return err
 	}
 
-	err = c.createTargetConfig(ctx, syncCtx.Recorder(), operatorSpec, envVars, etcd)
+	// Check if we need to remove the etcd container for external etcd usage.
+	// Currently used by DualReplica aka Two Node Fencing clusters.
+	shouldRemoveEtcdContainer := false
+	if c.externalEtcdClusterStatus.IsExternalEtcdCluster() && c.externalEtcdClusterStatus.IsReadyForEtcdTransition() {
```

Comment:
this could just be a ceohelper function, based on the input from v1helpers.StaticPodOperatorClient - that would also make your mocking 100% easier and it wouldn't be such a pain with the locking inside your status.

Plus points, you'd get the free informer cache synchronization from this controller